src/keys/ed25519.rs

```diff
@@ -33,9 +33,9 @@ impl EnrKey for ed25519::SigningKey {
             .ok_or(DecoderError::Custom("Unknown signature"))?;
 
         // Decode the RLP
-        let pubkey_bytes = rlp::Rlp::new(pubkey_bytes).data()?;
+        let pubkey_bytes = Bytes::decode(&mut pubkey_bytes.to_vec().as_slice())?;
 
-        Self::decode_public(pubkey_bytes)
+        Self::decode_public(pubkey_bytes.to_vec().as_slice())
```

Comment:
```suggestion
        Self::decode_public(&pubkey_bytes)
```
Saves another `Vec` allocation